### ISSUE TITLE
Shellcheck doesn't warn when using [ with bash

### ISSUE THREAD

**Issue by ntchambers** [koalaman/shellcheck#1105]
#### For bugs
- Rule Id: None
- My shellcheck version (`shellcheck --version` or "online"): 0.4.5 and online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

hi=foo

if [ "$hi" = bar ]; then
    :
fi
```

#### Here's what shellcheck currently says:
No output

#### Here's what I wanted or expected to see:
A warning about using [ over [[ with bash.

**Comment by Nightfirecat** [comment]
What is wrong with your snippet, exactly? `[` is perfectly valid in bash.

**Comment by ntchambers** [comment]
Shellcheck isn't about what is valid bash and what isn't. `echo $HOST` is valid bash, but it complains about unquoted variable expansions. `[[` is preferred over `[` in bash for a variety of reasons.

**Comment by ntchambers** [comment]
http://mywiki.wooledge.org/BashFAQ/031

**Comment by Nightfirecat** [comment]
Duplicate of #887. There's a lot of discussion on that issue, though it is still open.
